solutions/usaco-736.mdx

```diff
@@ -188,3 +188,75 @@ public class BovineGenomics {
 </LanguageSection>
 
 
+## Solution 3 - Unordered Map
+
+<LanguageSection>
+
+<CPPSection>
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(M * N)$
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int n, m, ans = 0;
+vector<string> gens;
+
+// Use `unordered_map` to achieve O(1) time complexity in queries
+unordered_map<char, int> spotties;
+
+int main()
+{
+  ios::sync_with_stdio(false);
+  cin.tie(0), cout.tie(0);
+
+  cin >> n >> m;
+  gens = vector<string>(2 * n);
+
+  for (int i = 0; i < 2 * n; i++)
+  {
+    cin >> gens[i];
+  }
+
+  for (int i = 0; i < m; i++)
+  {
+    // Let's clear characters history
+    spotties.clear();
+
+    for (int j = 0; j < 2 * n; j++)
+    {
+      char c = gens[j][i];
+
+      if (j < n)
+      {
+        // Spotty cows
+        // Check character `c` as seen
+        spotties[c]++;
+      }
+      else
+      {
+        // Plain cows
+        // Check if the gen has been seen on spotty cows
+        if (spotties.find(c) != spotties.end())
+          break;
+      }
+
+      // If the for-loop hasn't been broken yet, then we found another gen
+      if (j == 2 * n - 1)
```

Comment:
can you stay consistent and either use brackets or put it on the same line

Reply:
Sure